File: src/main/java/org/cerion/stocklist/web/CachedDataAPI.java
package org.cerion.stocklist.web;

import org.cerion.stocklist.Price;
import org.cerion.stocklist.model.Interval;

import java.util.Date;
import java.util.List;

public interface CachedDataAPI extends DataAPI {
    void clearCache();
    List<Price> getPrices(String symbol, Interval interval, Date start, boolean forceUpdate) throws Exception;
}

File: src/main/kotlin/web/CachedDataAPI.kt
package org.cerion.stocks.core.web

import org.cerion.stocks.core.PriceRow
import org.cerion.stocks.core.model.Interval
import java.util.*

interface CachedDataAPI : DataAPI {
    fun clearCache()

    @Throws(Exception::class)
    fun getPrices(symbol: String, interval: Interval, start: Date, forceUpdate: Boolean): List<PriceRow>
}
